2 ways masked numbers conversation

maylonpedroso opened the pull request:
- application frontend with numbers list and add form
- async phone number inactivity watcher

maylonpedroso committed:
Application frontend with numbers list and add form
Async phone number inactivity watcher

maylonpedroso committed:
Update project documentation

mrcrow85 commented on test/api_mocker.js:
```diff
@@ -0,0 +1,99 @@
+ * @param {object[]} options
+ * @return {function}
+ */
+class ApiMocker {
```
It seems to me we are reinventing the wheel here. Instead of this custom api mocker, lets use a proper mocking library like [sinon.js](https://sinonjs.org/).

mrcrow85 commented on test/conversations.spec.js:
```diff
@@ -0,0 +1,126 @@
+describe('conversations', function() {
+  describe('on create', function() {
+    it('makes a request and returns a ConversationInstance', async function() {
+      client.httpClient.request = mocker.mock([{ resource: 'Conversations' }]);
```
Here we can use stubs or mocks from `sinon` to fake the Twilio client. You can take a look at this [test case](https://github.com/TwilioDevEd/sample-pay-service/blob/master/test/services/payment-details.spec.js) for an example on how to work with sinon. In that case I used spies, but you can create mocks too.

mmena1 reviewed (changes requested):
Please take a look at the comments I left!

maylonpedroso commented on test/conversations.spec.js:
```diff
@@ -0,0 +1,126 @@
+const mocker = new ApiMocker(path.join(__dirname, 'fixtures'));
+
+describe('conversations', function() {
+  describe('on create', function() {
```
@mrcrow85 I am not sure there is a context here. It is just `describes` the method to create create a conversation.

mrcrow85 replied:
@maylonpedroso alright, fair enough. Then lets leave it like that, no worries.

maylonpedroso committed:
Refactor tests to use sinon to mock the API requets

mrcrow85 commented on test/conversations.spec.js:
```diff
@@ -0,0 +1,193 @@
+const sinon = require('sinon');
+const client = require('twilio')('ACxxx', 'fake_token');
+const conversations = require('../src/conversations/conversations');
+
```
To use `sinon` and `chai`, you have to tell chai to use the `sinon-chai` library:
```javascript
const sinonChai = require("sinon-chai");
const expect = chai.expect;
chai.use(sinonChai);
```
Maybe that's why the tests always pass regardless of using `calledOnce` or `calledTwice`?

maylonpedroso replied:
done! now it works. Thanks @mrcrow85

mrcrow85 commented on test/conversations.spec.js:
```diff
@@ -0,0 +1,193 @@
+const describe = require('mocha').describe;
```
This line isn't necessary. `describe` will be available at runtime when executing `npm test`

mrcrow85 commented on test/db.spec.js:
```diff
@@ -0,0 +1,128 @@
+const describe = require('mocha').describe;
```
Same here, not needed.

mmena1 reviewed (commented):
Good job! You are almost there, I left some minor comments.

mmena1 reviewed (approved):
Ok, LGTM! Good Job!

maylonpedroso committed:
Add instructions on how the application works to the dashboard.

jonedavis committed:
updated sample title

- Changed page title and h1 to "Masked Phone Numbers over SMS with Conversations"
- Added link to Twilio Conversations

jonedavis commented:
@maylonpedroso `npm run dev` causes a cyclical loop where nodemon is always restarting server. Can you take a 👀 ?

jonedavis committed:
Update README.md

jonedavis committed:
instructions tweaks

- punctuations on compound sentence
- removed Twilio from second use of "Conversations"

jonedavis committed:
Merge branch 'conversation' of https://github.com/TwilioDevEd/sample-conversations-masked-numbers into conversation

jonedavis committed:
update expiration message

Changed to "This message has expired!"

jonedavis commented:
Above commits have been around documentation.

maylonpedroso committed:
Ignore data files when running dev server deamon

jonedavis commented:
@dkundel can you 👀? Ready to publish and move to code-exchange.

dkundel commented on README.md:
```diff
@@ -18,35 +20,19 @@ Implementations in other languages:
-**TODO: Describe how it works**
--->
+This application creates a single conversation and provides a page to manage phone numbers in the conversation.
+Using a webhook it tracks the activity of every phone number in the conversation. When a phone number is iddle
```
There's a typo here. It should just be `idle`

dkundel commented on package.json:
```diff
@@ -1,25 +1,24 @@
   "repository": {
     "type": "git",
-    "url": "https://github.com/twilio-labs/sample-template-node"
+    "url": "https://github.com/TwilioDevEd/sample-conversations-masked-numbers"
```
This should be `twilio-labs` instead of `TwilioDevEd` as we'll move it there

jonedavis replied:
👍

dkundel commented on src/conversations/index.js:
```diff
@@ -0,0 +1,120 @@
+      conversation.sid,
+      number
+    );
+    return await new PhoneNumber({
```
the `await` here is unnecessary. It anyways returns a Promise in either way

dkundel replied:
Applies throughout

jonedavis replied:
👍

dkundel commented on views/phone_form.pug:
```diff
@@ -0,0 +1,12 @@
+      input#phone-number.form-control(name='phoneNumber' required='required')
+      label(for='expiration-time') Expiration Time
+      input#expiration-time.form-control(name='expirationTime' required='required')
+      small.form-text.text-muted Max time allowed to be inactive in minutes before been removed from the conversatiosn
```
There's a typo here. It's `conversations` not `conversatiosn`

jonedavis replied:
👍

jonedavis committed:
quality pass

- fixed spelling error
- remove unneeded await
- repo points to future home

dkundel committed:
docs: fix typo
